src/transform-namespaces.test.ts.md

```diff
@@ -0,0 +1,143 @@
+# Snapshot report for `src/transform-namespaces.test.ts`
+
+The actual snapshot is saved in `transform-namespaces.test.ts.snap`.
+
+Generated by [AVA](https://avajs.dev).
+
+## add namespace
+
+> Snapshot 1
+
+    `// SPDX-License-Identifier: UNLICENSED␊
+    pragma solidity ^0.8.20;␊
+    ␊
+    contract C1 {␊
+        function __C1_init() internal onlyInitializing {␊
+        }␊
+    ␊
+        function __C1_init_unchained() internal onlyInitializing {␊
+        }␊
+    }␊
+    ␊
+    contract C2 {␊
+        function __C2_init() internal onlyInitializing {␊
+            __C2_init_unchained();␊
+        }␊
+    ␊
+        function __C2_init_unchained() internal onlyInitializing {␊
+            C2Storage storage $ = _getC2Storage();␊
+            $.s1 = "";␊
+        }␊
+        event B();␊
+    ␊
+        struct A { uint z; }␊
+    ␊
+        uint constant C = 3;␊
+        // b␊
+        /// @custom:oz-upgrades-unsafe-allow state-variable-immutable␊
+        uint immutable y = 2;␊
+    ␊
+        /// @custom:storage-location erc7201:openzeppelin.storage.C2␊
+        struct C2Storage {␊
+            // a␊
+            uint x;␊
+            uint z;␊
+    ␊
+            string s1;␊
+        }␊
```

Comment:
spacing is a bit strange.

Reply:
Spacing here is inherited from the original code, where variables look like this:

```solidity
    uint constant C = 3;
    // a
    uint x;
    // b
    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
    uint immutable y = 2;
    uint private z;

    string private s1 = "";
```

So I think it's good!

Reply:
It really comes down to what `// a` and `// b` means and refer to. I find it strange that `// a` is attached to x, and `// b` is attached to `y`.

Reply:
that may not be an issue if variables are sorted (constants, then immutable, then private)

Reply:
My reasoning was that if a comment is immediately before a variable, it should be attached to that variable. But I didn't consider the case where there is a newline between comment and variable, in that case I would say they should be considered separate and the comment should possibly be left outside of the struct...

```
// atached comment
uint y;

// separate comment

uint x;
```

Reply:
We're very limited in how we can handle comments because they're not present in the AST so I would leave the current behavior, unless it's producing results that we don't like for the Contracts codebaes.